mapfile.c

```diff
@@ -424,9 +452,13 @@ int loadColor(colorObj *color, attributeBindingObj *binding)
 
   color->alpha=255;
   if(symbol == MS_NUMBER) {
-    color->red = (int) msyynumber;
-    if(getInteger(&(color->green)) == -1) return MS_FAILURE;
-    if(getInteger(&(color->blue)) == -1) return MS_FAILURE;
+    if(msyynumber >= -255 && msyynumber <= 255) {
```

Comment:
why do we accept negative values for colors ?

Reply:
It's used to trigger alternate behavior in at least one case. https://mapserver.org/mapfile/reference.html says that a negative color component is used to draw an outline and no fill. We could add a flag to getColor() to allow negative color components and only set that to true for the reference color. We do initialize colors to -1,-1,-1 in a number of spots.

Reply:
putting the above as a comment in the code would be helpful :-)

Reply:
You mean part of my explanation?

Reply:
yes, it is not obvious when reading the code why a negative value for a color is accepted

Reply:
added comment...